typescript/utils/src/altvm.ts

```diff
@@ -0,0 +1,657 @@
+import { MINIMUM_GAS } from './mingas.js';
+import { ProtocolType } from './types.js';
+
+// ### QUERY BASE ###
+export type ReqGetBalance = { address: string; denom: string };
+
+export type ReqGetTotalSupply = { denom: string };
+
+export type ReqEstimateTransactionFee<T> = {
+  transaction: T;
+  estimatedGasPrice: string;
+  sender: string;
+  senderPubKey?: string;
```

Comment:
what is the difference between `sender` and `senderPubKey`?

Reply:
sender is the sender address (will rename to `senderAddress`) while `senderPubKey` is the public key, on some protocols these are different and the public key can't be derived from the address